### PR TITLE
Fix secret updater

### PR DESCRIPTION
**Pull request by kmdkuk**
#34 

- The secret updater is implemented to periodically check if the secret has expired every few seconds.
- The secret updater has been modified to check and update the secret only just before it expires.

Signed-off-by: kouki <kouworld0123@gmail.com>

### CODE REVIEW

File: controllers/secret_updater.go [hunk 1]
```diff
@@ -7,96 +7,213 @@ import (
...
 }
 
-// Start implements Runnable.Start
-func (u SecretUpdater) Start(ctx context.Context) error {
-	ticker := time.NewTicker(u.interval)
-	defer ticker.Stop()
-	logger := log.FromContext(ctx).WithName("SecretUpdater")
+func newUpdateProcess(log logr.Logger, client client.Client, githubClient github.Client, repositoryName, rpNamespacedNameStr string, secretNamespacedName types.NamespacedName) *updateProcess {
+	return &updateProcess{
+		log:                  log.WithValues("RunnerPoolName", rpNamespacedNameStr),
```
Comment (masa213f) [inline controllers/secret_updater.go:85]: Please use the same field name.
https://github.com/cybozu-go/meows/blob/78b88a57f863794e032455e579fe30a78060db72/controllers/runnerpool_controller.go#L61

```suggestion
		log:                  log.WithValues("runnerpool", rpNamespacedNameStr),
```

File: docs/metrics.md [hunk 2]
```diff
@@ -6,19 +6,20 @@ Metrics
...
 Aside from [the standard Go runtime and process metrics][standard], it exposes metrics related to controller-runtime and RunnerPools.
 
-| Name                          | Description                            | Type  | Labels                 |
-| ----------------------------- | -------------------------------------- | ----- | ---------------------- |
-| `meows_runnerpool_replicas` | The number of the RunnerPool replicas. | Gauge | `runnerpool`           |
-| `meows_runner_online`       | 1 if the runner is online.             | Gauge | `runnerpool`, `runner` |
-| `meows_runner_busy`         | 1 if the runner is busy.               | Gauge | `runnerpool`, `runner` |
+| Name                           | Description                                                        | Type  | Labels                 |
+| ------------------------------ | ------------------------------------------------------------------ | ----- | ---------------------- |
+| `meows_runnerpool_secret_retry_count` | The number of times meows retried continuously to get github token | Gauge | `runnerpool`           |
```
Comment (masa213f) [inline docs/metrics.md:11]: Please correct the formatting.

**Review by masa213f — APPROVED** [review]
LGTM

File: controllers/runnerpool_controller_test.go [hunk 3]
```diff
@@ -587,41 +564,60 @@ var _ = Describe("RunnerPool reconciler", func() {
...
+			githubFakeClient.ExpiredAtDuration = 1 * time.Hour
+
+			time.Sleep(10 * time.Second)
 
 			if tc.shouldUpdate {
 				By("checking to update secret")
 				s = new(corev1.Secret)
 				err := k8sClient.Get(ctx, types.NamespacedName{Name: secretName, Namespace: namespace}, s)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).ShouldNot(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (zoetrope) [reply controllers/runnerpool_controller_test.go]: I don't understand your intentions.

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**